<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>675</width>
    <height>455</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Rubics cube</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../Робочий стіл/cube.png</normaloff>../Робочий стіл/cube.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>130</y>
      <width>181</width>
      <height>141</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout"/>
   </widget>
   <widget class="QLabel" name="testoutput">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Your Hint will show here</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="InputRotateName">
    <property name="geometry">
     <rect>
      <x>187</x>
      <y>41</y>
      <width>221</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>120</y>
      <width>401</width>
      <height>251</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2"/>
   </widget>
   <widget class="QSlider" name="speedSlider">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>280</y>
      <width>171</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>4</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>191</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>min             speed               max</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>110</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Hint Window</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>142</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Input your command:</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>21</x>
      <y>10</y>
      <width>633</width>
      <height>29</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="RandomButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>R</string>
       </property>
       <property name="statusTip">
        <string>R</string>
       </property>
       <property name="text">
        <string>Random</string>
       </property>
       <property name="shortcut">
        <string>R</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ShowHintButton">
       <property name="toolTip">
        <string>S</string>
       </property>
       <property name="statusTip">
        <string>S</string>
       </property>
       <property name="text">
        <string>ShowHint</string>
       </property>
       <property name="shortcut">
        <string>S</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SelectUPAxisButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>T</string>
       </property>
       <property name="statusTip">
        <string>T</string>
       </property>
       <property name="text">
        <string>Top</string>
       </property>
       <property name="shortcut">
        <string>T</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SelectFDAxisButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>F</string>
       </property>
       <property name="statusTip">
        <string>F</string>
       </property>
       <property name="text">
        <string>Front</string>
       </property>
       <property name="shortcut">
        <string>F</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="SelectSDAxisButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>D</string>
       </property>
       <property name="statusTip">
        <string>D</string>
       </property>
       <property name="text">
        <string>Side</string>
       </property>
       <property name="shortcut">
        <string>D</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="PreviousLayerButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Up</string>
       </property>
       <property name="statusTip">
        <string>Up</string>
       </property>
       <property name="text">
        <string>UP</string>
       </property>
       <property name="shortcut">
        <string>Up</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="NextLayerButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Down</string>
       </property>
       <property name="statusTip">
        <string>Down</string>
       </property>
       <property name="text">
        <string>Down</string>
       </property>
       <property name="shortcut">
        <string>Down</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>40</y>
      <width>178</width>
      <height>29</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="R_Rotatebutton">
       <property name="toolTip">
        <string>G</string>
       </property>
       <property name="statusTip">
        <string>G</string>
       </property>
       <property name="text">
        <string>Go</string>
       </property>
       <property name="shortcut">
        <string>G</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="CounterRotate">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>space</string>
       </property>
       <property name="statusTip">
        <string>space</string>
       </property>
       <property name="text">
        <string>Rotate</string>
       </property>
       <property name="shortcut">
        <string>Space</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>70</y>
      <width>221</width>
      <height>29</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QPushButton" name="finishhintbutton">
       <property name="toolTip">
        <string>H</string>
       </property>
       <property name="statusTip">
        <string>H</string>
       </property>
       <property name="text">
        <string>Finish Hint steps</string>
       </property>
       <property name="shortcut">
        <string>H</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="RotateButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>27</height>
        </size>
       </property>
       <property name="toolTip">
        <string>C</string>
       </property>
       <property name="statusTip">
        <string>C</string>
       </property>
       <property name="text">
        <string>Rotate'</string>
       </property>
       <property name="shortcut">
        <string>C</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QCheckBox" name="singmastercheck">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>330</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>SingMaster Method</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="Oppocrosscheck">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>360</y>
      <width>171</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>OppoCross Method</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>675</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionSave"/>
    <addaction name="actionLoad"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
    <addaction name="actionAuthor"/>
    <addaction name="actionAbout_the_hint"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionSave_Image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionAuthor">
   <property name="text">
    <string>Author</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionAbout_the_hint">
   <property name="text">
    <string>About the hint</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>finishhintbutton</sender>
   <signal>clicked()</signal>
   <receiver>ShowHintButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>SaveImageToFile()</slot>
 </slots>
</ui>
